watchdog/types/types.go

```diff
@@ -44,3 +55,19 @@ func MarshalRequest(data []byte, header *http.Header) ([]byte, error) {
 	res, marshalErr := json.Marshal(&req)
 	return res, marshalErr
 }
+
+func UnmarshalForm(data []byte) (*Form, error) {
+	form := new(Form)
+	err := json.Unmarshal(data, form)
+	return form, err
+}
+
+func MarshalForm(data map[string]FormPart, header *http.Header) ([]byte, error) {
+	form := Form{
```

Comment:
Is this standard, or ad-hoc?

Reply:
If you mean the Marshal/Unmarshal functions, I'd added them to keep the Form data handling the same as the existing `marshal_request` handling.